core/deployment/src/main/java/io/quarkus/deployment/util/ProcessUtil.java

```diff
@@ -0,0 +1,138 @@
+package io.quarkus.deployment.util;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.deployment.pkg.builditem.ProcessInheritIODisabled;
+
+/**
+ * Utility for {@link Process} related operations
+ */
+public class ProcessUtil {
+
+    private static final Logger logger = Logger.getLogger(ProcessUtil.class);
+
+    /**
+     * Launches and returns a {@link Process} built from the {@link ProcessBuilder builder}.
+     * Before launching the process, this method checks if {@link ProcessInheritIODisabled inherit IO is disabled}
+     * and if so, streams both the {@code STDOUT} and {@code STDERR} of the launched process using
+     * {@link #streamToSysOutSysErr(Process)}. Else, it launches the process with {@link ProcessBuilder#inheritIO()}
+     *
+     * @param builder The process builder
+     * @param processInheritIODisabled Whether or not {@link java.lang.ProcessBuilder.Redirect#INHERIT} can be used for
+     *        launching the process
+     * @return Returns the newly launched process
+     * @throws IOException
+     */
+    public static Process launchProcess(final ProcessBuilder builder,
+            final Optional<ProcessInheritIODisabled> processInheritIODisabled) throws IOException {
+        if (!processInheritIODisabled.isPresent()) {
+            return builder.inheritIO().start();
+        }
+        final Process process = builder.redirectOutput(ProcessBuilder.Redirect.PIPE)
+                .redirectError(ProcessBuilder.Redirect.PIPE)
+                .start();
+        // stream both stdout and stderr of the process
+        ProcessUtil.streamToSysOutSysErr(process);
```

Comment:
Aren't standard and error streams merged into one in this case?

Reply:
Not unless the `ProcessBuilder#redirectErrorStream(boolean)` method is called.